tests/testImageDecorrelation.py

```diff
@@ -190,65 +182,124 @@ def setUp(self):
+        # Uncorrected diffim exposure - variance plane is wrong (too low)
+        statObj = afwMath.makeStatistics(diffExp.getMaskedImage().getVariance(),
+                                         diffExp.getMaskedImage().getMask(), afwMath.MEANCLIP,
+                                         self.statsControl)
+        mn = statObj.getValue(afwMath.MEANCLIP)
+        self.assertNotClose(mn, expected_var, rtol=0.02)
+        print('UNCORRECTED VARIANCE:', var, mn)
 
         task = DecorrelateALKernelTask()
-        decorrResult = task.run(self.im1ex, self.im2ex, diffExp, mKernel)
+        mKernelPsf = measAlg.KernelPsf(mKernel)
+        decorrResult = task.run(self.im1ex, self.im2ex, diffExp, mKernelPsf)
         corrected_diffExp = decorrResult.correctedExposure
 
         # Corrected diffim - variance should be close to expected.
-        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage(), afwMath.VARIANCE,
+        # We set the tolerance a bit higher here since the simulated images have many bright stars
+        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage(), afwMath.VARIANCECLIP,
                                          self.statsControl)
-        var = statObj.getValue(afwMath.VARIANCE)
-        print(var, expected_var)
-        self.assertClose(var, expected_var, rtol=0.02)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+        self.assertClose(var, expected_var, rtol=0.05)
+
+        # Check statistics of variance plane in corrected diffim
+        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage().getVariance(),
+                                         corrected_diffExp.getMaskedImage().getMask(),
+                                         afwMath.MEANCLIP, self.statsControl)
+        mn = statObj.getValue(afwMath.MEANCLIP)
+        print('CORRECTED VARIANCE:', var, mn)
+        self.assertClose(mn, expected_var, rtol=0.05)
+        self.assertClose(var, mn, rtol=0.05)
+
+    def testDiffimCorrection1(self):
+        """! Test two images with the same variance.
+
+        Check that the variance of the corrected diffim matches the theoretical value
+        (to within a 1% tolerance). Use
+        """
+        self._setUpImages(var1=0.04, var2=0.04)
+        self._testImages()
+
+    def testDiffimCorrection2(self):
+        """! Test when the science image variance is higher than that of the template.
+
+        Check that the variance of the corrected diffim matches the theoretical value
+        (to within a 1% tolerance).
+        """
+        self._setUpImages(var1=0.08, var2=0.04)
+        self._testImages()
+
+    def testDiffimCorrection3(self):
+        """! Test when the template variance is higher than that of the science img.
+
+        Check that the variance of the corrected diffim matches the theoretical value
+        (to within a 1% tolerance).
+        """
+        self._setUpImages(var1=0.04, var2=0.08)
+        self._testImages()
 
 
 def suite():
```

Comment:
TestSuite is dead: long live py.test! See SQR-012 and the developer docs for more details:

https://sqr-012.lsst.io/
https://developer.lsst.io/coding/python_testing.html